src/library_manager/lib_manager.c

```diff
@@ -444,23 +444,28 @@ static int lib_manager_dma_deinit(struct lib_manager_dma_ext *dma_ext, uint32_t
 
 static int lib_manager_load_data_from_host(struct lib_manager_dma_ext *dma_ext, uint32_t size)
 {
+	uint64_t timeout = k_ms_to_cyc_ceil64(200);
 	struct dma_status stat;
 	int ret;
 
-	ret = dma_get_status(dma_ext->chan->dma->z_dev, dma_ext->chan->index, &stat);
-	if (ret < 0)
-		return ret;
+	/* Wait till whole data acquired with timeout of 200ms */
+	timeout += sof_cycle_get_64();
 
-	/* Wait till whole data acquired */
-	while (stat.pending_length < size) {
-		k_usleep(100);
-
-		ret = dma_get_status(dma_ext->chan->dma->z_dev, dma_ext->chan->index, &stat);
-		if (ret < 0)
+	for (;;) {
+		ret = dma_get_status(dma_ext->chan->dma->z_dev,
+				     dma_ext->chan->index, &stat);
+		if (ret < 0 || stat.pending_length >= size)
 			return ret;
+
+		if (sof_cycle_get_64() > timeout)
```

Comment:
on the first iteration this *really* shouldn't trigger - we shouldn't be spending more than 200ms between line 452 and here, so I think the previous version with
```
    k_usleep(100);
} while (sof_cycle_get_64() <= timeout);
```
was good too :-) But not too important